src/App.jsx

```diff
@@ -14,33 +17,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [showReversed, setShowReversed] = useState(false);
+  const [sortedBy, setSortedBy] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const applyReverse = data => (showReversed ? data.reverse() : data);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const processData = (data, sort) => {
+    setVisibleGoods(applyReverse(data));
+    setSortedBy(sort);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortByAlphabet = () => {
+    processData(
+      [...goodsFromServer].sort((a, b) => a.localeCompare(b)),
+      'alphabet',
+    );
+  };
+
+  const sortByLength = () => {
```

Comment:
In your case, it looks too complicated. You can create a simple helper function outside the component that sort goods and returns them prepared to render